examples/nature_of_code/chp_01_vectors/1_10_motion101_acceleration.rs

```diff
@@ -0,0 +1,90 @@
+// The Nature of Code
+// Daniel Shiffman
+// http://natureofcode.com
+//
+// Example 1-10: Motion 101 Acceleration
+extern crate nannou;
+
+use nannou::prelude::*;
+
+fn main() {
+    nannou::app(model, event, view).run();
+}
+
+struct Model {
+    mover: Mover,
+}
+
+struct Mover {
+    position: Point2<f32>,
+    velocity: Vector2<f32>,
+    acceleration: Vector2<f32>,
+    top_speed: f32,
+}
+
+impl Mover {
+    fn new(rect: Rect<f32>) -> Self {
+        let position = Point2::new(rect.x(), rect.y());
+        let velocity = Vector2::new(0.0, 0.0);
+        let acceleration = Vector2::new(0.0, 0.0);
+        let top_speed = 5.0;
+        Mover {
+            position,
+            velocity,
+            acceleration,
+            top_speed,
+        }
+    }
+
+    fn update(&mut self, mouse: Point2<f32>) {
+        // Computer a vector that points from position to mouse
+        self.acceleration = mouse - self.position;
+        // Set magnitude of acceleration
+        self.acceleration = self.acceleration.normalize() * 0.2;
+        // Velocity chages according to acceleration
+        self.velocity += self.acceleration;
+        // Limit the velocity by top_speed
+        self.velocity = vec2(
+            self.velocity.x.min(self.top_speed),
+            self.velocity.y.min(self.top_speed),
+        );
```

Comment:
This is not quite the same as the `limit` method in processing. `limit` clamps the *magnitude* rather than the value of both axes. Acceleration might behave faster when moving in diagonal directions.